### PR TITLE
[Consensus] Dedicated channel for proposal processing

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

We have seen cases where proposals are being processed after the QS for that proposal, which causes the nodes to fetch the block from remote. This change dedicates a channel for proposal, so that we can prioritize processing proposals above other messages. Please note that it can still happen that the proposals arrive out of order - we need to see how often that happens and re-evaluate the ordering if needed. 

### Test Plan

Run Forge

### CODE REVIEW

File: consensus/src/round_manager.rs [hunk 1]
```diff
@@ -977,26 +982,54 @@ impl RoundManager {
...
+        mut checked_proposal_rx: tokio::sync::mpsc::Receiver<VerifiedEvent>,
         close_rx: oneshot::Receiver<oneshot::Sender<()>>,
     ) {
         info!(epoch = self.epoch_state().epoch, "RoundManager started");
         let mut close_rx = close_rx.into_stream();
         loop {
-            futures::select! {
+            tokio::select! {
+                biased;
+                close_req = close_rx.select_next_some() => {
```
Comment (bchocho) [inline consensus/src/round_manager.rs:993]: Cool. I wonder if shutdown will be faster?

**Review by bchocho — COMMENTED** [review]
Overall looks good. Looks like some tests can't compile

File: consensus/src/round_manager.rs [hunk 2]
```diff
@@ -977,26 +982,54 @@ impl RoundManager {
...
+                    while let Ok(proposal) = checked_proposal_rx.try_recv() {
+                        proposals.push( proposal);
+                    }
+                    proposals.sort_by(|a, b| a.round().unwrap().cmp(&b.round().unwrap()));
+                    for proposal in proposals {
+                        let result = match proposal {
+                            VerifiedEvent::ProposalMsg(proposal_msg) => {
+                                monitor!(
+                                    "process_proposal",
+                                    self.process_proposal_msg(*proposal_msg).await
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/round_manager.rs:1010]: fast forward sync will sync us to the remote highest cert positions including highest ordered cert (by fetching blocks)

**Review by zekun000 — APPROVED** [review]

**Comment by davidiw** [comment]
stale?

**Pull request closed without merging**